cfme/fixtures/candu.py

```diff
@@ -90,3 +123,367 @@ def candu_db_restore(temp_appliance_extended_db):
+        rollup_subq = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly',
+                 rollups.resource_name == vm_name,
+                 ems.name == provider.name,
+                 rollups.timestamp >= date.today())
+            .subquery()
+        )
+
+        for record in appliance.db.client.session.query(rollups).filter(
+                rollups.id.in_(rollup_subq)):
+            consumed_hours = consumed_hours + 1
+            cpu_used = cpu_used + getattr(record, 'cpu_usagemhz_rate_average', 0)
+            memory_used = memory_used + getattr(record, 'derived_memory_used', 0)
+            network_used = network_used + getattr(record, 'net_usage_rate_average', 0)
+            disk_used = disk_used + getattr(record, 'disk_usage_rate_average', 0)
+            storage_used = storage_used + getattr(record, 'derived_vm_used_disk_storage', 0)
+
+            if cpu_alloc == 0:
+                cpu_alloc = getattr(record, 'derived_vm_numvcpus', 0)
+            if memory_alloc == 0:
+                memory_alloc = getattr(record, 'derived_memory_available', 0)
+            if storage_alloc == 0:
+                storage_alloc = (getattr(record, 'derived_vm_allocated_disk_storage', 0)
+                    * math.pow(2, -30))
+
+    storage_used = storage_used * math.pow(2, -30)
+
+    yield dict(
+        cpu_used=cpu_used,
+        memory_used=memory_used,
+        network_used=network_used,
+        disk_used=disk_used,
+        storage_used=storage_used,
+        consumed_hours=consumed_hours,
+        cpu_alloc=cpu_alloc,
+        memory_alloc=memory_alloc,
+        storage_alloc=storage_alloc
+    )
+
+    appliance.server.settings.enable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector')
+
+
+def resource_cost(appliance, metric_description, usage, rate_description, rate_type,
+        consumed_hours):
+    """Query the DB for chargeback rates."""
+    cb_rates = appliance.db.client['chargeback_rates']
+    cb_rate_details = appliance.db.client['chargeback_rate_details']
+    cb_tiers = appliance.db.client['chargeback_tiers']
+
+    field_map = {
+        1: ['per_time'],
+        2: ['variable_rate', 'fixed_rate', 'start', 'finish']
+    }
+    rates = []
+    rate_tier = {}
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(cb_rates, cb_rate_details, cb_tiers)
+            .filter(cb_rates.id == cb_rate_details.chargeback_rate_id)
+            .filter(cb_rate_details.id == cb_tiers.chargeback_rate_detail_id)
+            .filter(cb_rates.description == rate_description)
+            .filter(cb_rates.rate_type == rate_type)
+            .filter(cb_rate_details.description == metric_description)
+            .all()
+        )
+        for row in result:
+            for index in field_map:
```

Comment:
Consider using `enumerate(<iterable>, start=1)` instead of creating a dictionary with integer indices.

```
field_map = [
    ['per_time'],
    ['variable', 'fixed', 'start', 'finish']
]

for row in result:
    for index, fields in enumerate(field_map, start=1):
        rate_tier.update({
            var: getattr(row[index], var) for var in fields
        })
```